engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/SchemaRegistryProvider.scala

```diff
@@ -1,17 +1,18 @@
 package pl.touk.nussknacker.engine.avro.schemaregistry
 
 import pl.touk.nussknacker.engine.kafka.RecordFormatter
-import pl.touk.nussknacker.engine.kafka.serialization.{DeserializationSchemaFactory, SerializationSchemaFactory}
+import pl.touk.nussknacker.engine.kafka.serialization.{KafkaDeserializationSchemaVersionAwareFactory, KafkaSerializationSchemaVersionAwareFactory}
 
 /**
   * @tparam T - Scheme used to deserialize
   */
 trait SchemaRegistryProvider[T] extends Serializable {
+
   def createSchemaRegistryClient: SchemaRegistryClient
 
-  def deserializationSchemaFactory: DeserializationSchemaFactory[T]
+  def deserializationSchemaFactory: KafkaDeserializationSchemaVersionAwareFactory[T]
```

Comment:
Maybe better will be `KafkaVersionAwareDeserializationSchemaFactory`?
Props: It will be ended up by `DeserializationSchemaFactory` and will be easy to recognize that it is factory for Flink's `DeserializationSchema`

Reply:
You know me..  - I will change it ;)